_episodes/l4_04_parameterized_tests.md

```diff
@@ -0,0 +1,353 @@
+---
+title: "Parameterised Tests using GoogleTest"
+teaching: TBC
+exercises: TBC
+questions:
+- "What is a parameterised test?"
+- "How to write your own parameterised tests using GoogleTest?"
+objectives:
+- "Understand why do we need a parameterised test."
+- "Understand how to create a parameterised test using GoogleTest."
+- "Understand the advantages of parameterised tests."
+- "Understand how to combine a test fixtures with parameterised test."
+keypoints:
+- "We learnt about parameterised tests in GoogleTest."
+- "We learnt how to create our own parameterised tests."
+- "We learnt how to combine test fixture with parameterised test."
+---
+
+## 1. Introduction to Parameterised Tests
+
+**Parameterized tests**, also known as **data-driven tests**, are a feature provided by testing frameworks like Google Test that allows us to write a single test case that can be executed with different sets of test data or inputs. Instead of duplicating similar test cases with slight variations, parameterized tests enable us to define a test once and run it with multiple inputs or test data.
```

Comment:
```suggestion
**Parameterised tests**, also known as **data-driven tests**, are a feature provided by testing frameworks like Google Test that allows us to write a single test case that can be executed with different sets of test data or inputs. Instead of duplicating similar test cases with slight variations, parameterised tests enable us to define a test once and run it with multiple inputs or test data.
```